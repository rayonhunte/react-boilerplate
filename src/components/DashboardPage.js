import React from 'react';


const DashboardPage = () => (
  <div>
    this is the bash
  </div>
);

export default DashboardPage;