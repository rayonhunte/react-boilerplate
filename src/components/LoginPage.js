import React from 'react';
import {connect} from 'react-redux';
import { startLogin } from '../actions/auth';

export const  LoginPage = ({startLogin})=> {
  return (
      <div className="box-layout">
          <div className="box-layout__box">
            <h1 className="layout__title">React Boilerplate</h1>
            <p>Tag Line For App !!!</p>
            <button onClick={startLogin} className="button">Login With Google</button>
          </div>
      </div>
    );
  
};

const mapDispatchToProps = (dispatch)=>({
  startLogin: ()=> dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);