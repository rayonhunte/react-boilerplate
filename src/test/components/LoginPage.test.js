import React from 'react';
import {shallow} from 'enzyme';
import {LoginPage} from '../../components/LoginPage';


let startLogin, wrapper;


beforeEach(()=>{
  startLogin = jest.fn();
  wrapper = shallow(<LoginPage startLogin={startLogin}/>);
});


test('should render login', ()=>{
  expect(wrapper).toMatchSnapshot();
});





test('test login',()=>{
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalledWith();  
});