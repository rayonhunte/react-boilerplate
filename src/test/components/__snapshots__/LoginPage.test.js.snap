// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="layout__title"
    >
      React Boilerplate
    </h1>
    <p>
      Tag Line For App !!!
    </p>
    <button
      className="button"
      onClick={[Function]}
    >
      Login With Google
    </button>
  </div>
</div>
`;
